iceoryx_platform/qnx/include/iceoryx_platform/attributes.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_QNX_PLATFORM_ATTRIBUTES_HPP
+#define IOX_HOOFS_QNX_PLATFORM_ATTRIBUTES_HPP
+
+/// @todo iox-#638
+#define IOX_NO_DISCARD
+
+/// @brief IOX_FALLTHROUGH adds the [[fallthrough]] keyword when it is available for the current compiler.
+/// @note
+//    [[fallthrough]] supported since gcc 7 (https://gcc.gnu.org/projects/cxx-status.html)
+///   [[fallthrough]] supported since clang 3.9 (https://clang.llvm.org/cxx_status.html)
+///   activate keywords for gcc>=7 or clang>=4
+
+#if __cplusplus >= 201703L
```

Comment:
On QNX we support only C++14 and gcc, therefore it should be only:

```cxx
#define IOX_FALLTHROUGH
#define IOX_MAYBE_UNUSED [[gnu::unused]]
```

Reply:
@elfenpiff I left the check for the C++ version here, as in some hopefully not too distant future we will move to C++17. Although, other folks might be using a newer QCC version. No need to limit things here.